bin/utils/node-utils.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/sh
+
+################################################################################
+# This program and the accompanying materials are made available under the terms of the
+# Eclipse Public License v2.0 which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright IBM Corporation 2020
+################################################################################
+
+# TODO LATER - anyway to do this better?
+# Try and work out where we are even if sourced
+if [[ -n ${INSTALL_DIR} ]]
+then
+  export utils_dir="${INSTALL_DIR}/bin/utils"
+elif [[ -n ${ZOWE_ROOT_DIR} ]]
+then
+  export utils_dir="${ZOWE_ROOT_DIR}/bin/utils"
+elif [[ -n ${ROOT_DIR} ]]
+then
+  export utils_dir="${ROOT_DIR}/bin/utils"
+elif [[ $0 == "node-utils.sh" ]] #Not called by source
+then
+  export utils_dir=$(cd $(dirname $0);pwd)
+else
+  echo "Could not work out the path to the utils directory. Please 'export ZOWE_ROOT_DIR=<zowe-install-directory' before running." 1>&2
+  exit 1
+fi
+
+# Source common util functions
+. ${utils_dir}/common.sh
+
+# TODO - how to test well given interaction and guess?
+# Interactive function that checks if the current NODE_HOME is valid and if not requests a user enters the node home path via command line
+prompt_node_home() {
+  # If NODE_HOME not set, guess a default value
+  if [[ -z ${NODE_HOME} ]]
+  then
+    NODE_HOME="/usr/lpp/IBM/cnj/IBM/node-latest-os390-s390x"
+  fi
+  loop=1
+  while [ ${loop} -eq 1 ]
+  do
+    loop=0 # only want to re-run if user re-prompts
+    validate_node_home # Note - this outputs messages for errors found
+    node_valid_rc=$?
+    if [[ ${node_valid_rc} -ne 0 ]]
+    then
+      echo "Press Y or y to accept current node home '${NODE_HOME}', or Enter to choose another location"
+      read rep
+      if [ "$rep" != "Y" ] && [ "$rep" != "y" ]
+      then
+        echo "Please enter a path to where node is installed.  This is the a directory that contains /bin/node "
+        read NODE_HOME
+        loop=1
+      fi
+    fi
+  done
+  export NODE_HOME=$NODE_HOME
+  echo "  NODE_HOME variable value="${NODE_HOME} >> $LOG_FILE
+}
+
+ensure_node_is_on_path() {
+  if [[ ":$PATH:" != *":$NODE_HOME/bin:"* ]]
+  then
+    echo "Appending NODE_HOME/bin to the PATH..."
+    export PATH=$PATH:$NODE_HOME/bin
+  fi
+}
+
+validate_node_home() {
+  validate_node_home_not_empty
+  node_empty_rc=$?
+  if [[ ${node_empty_rc} -ne 0 ]]
+  then
+    return ${node_empty_rc}
+  fi
+
+  ls ${NODE_HOME}/bin | grep node$ > /dev/null
+  if [[ $? -ne 0 ]];
+  then
+    print_error_message "NODE_HOME: ${NODE_HOME}/bin does not point to a valid install of Node"
+    return 1
+  fi
+
+  node_ok=`${NODE_HOME}/bin/node -e "console.log('ok')" 2>&1`
```

Comment:
This line has a chance to hang in a rare case. So maybe we echo something before this line to indicate we are validating if node bin is functional. So when we checking the log, we are sure the flow hangs on this line.

Also seems better this section moved below `node -v` check.

Reply:
I've added the logs and moved this check below the version, which required some refactoring in order to still test